src/Microsoft.Diagnostics.Monitoring.WebApi/Validation/CounterValidator.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.WebApi.Models;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Globalization;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi.Validation
+{
+    internal static class CounterValidator
+    {
+        public static bool ValidateProvider(GlobalCounterOptions counterOptions,
+            EventPipeProvider provider,
+            out string errorMessage)
+        {
+            errorMessage = null;
+
+            if (provider.Arguments?.TryGetValue("EventCounterIntervalSec", out string intervalValue) == true)
+            {
+                if (int.TryParse(intervalValue, out int intervalSeconds) &&
```

Comment:
The event counters infrastructure in the runtime accepts values that are parsable as `float`. I think we should do the same here (`float.TryParse`) so that a non-whole integer interval (e.g. 0.5) in the provider configuration doesn't get past this validation. Later, we can update the GlobalCounter:IntervalSeconds property and underlying pipelines to take a `float` instead of an `int` to be consistent with the runtime.